### PR TITLE
Comments: Fix blank post title

### LINKED ISSUE #18993: Comments: Consider unicode whitespace characters when checking for missing post titles
See: https://github.com/Automattic/wp-calypso/pull/18884#pullrequestreview-70426208

> If we have unicode whitespace characters we'll still get a visual blank. the RegExp `\s` though I believe takes those into consideration, so if we wanted to iterate…

```js
const getTitle = ( post, comment ) => {
	const ws = /^\s*$/;
	const title = decodeEntities( get( comment, 'post.title' ) );
	if ( ! ws.test( title ) ) {
		return title;
	}

	// might `post` be `undefined`?
	// if not, we can just use `post.excerpt`
	// and leave out the `get()`
	const excerpt = stripHTML( get( post, 'excerpt' ) );
	if ( ws.test( excerpt ) ) {
		return excerpt;
	}

	return translate( 'Untitled Post' );
}
```

### LINKED ISSUE #19486: Comments: untitled posts create display issues
screen shot 2017-11-03 at 2 20 31 pm

Notice how some of these comments are on untitled posts (photo posts). For some reason, some display `Untitled` some are just blank? I don't know why it's blank or Untitled in some places.

Notice the title in the single post view:

screen shot 2017-11-03 at 2 23 10 pm

And just blank while expanded:

screen shot 2017-11-03 at 2 24 38 pm

### PR DESCRIPTION
**Pull request by Copons**
Fixes #18993 
Tentatively fixes #19486 (I can't seem to be able to replicate it)

By just `String.prototype.trim()`-ming the `postTitle` we remove both spaces and Unicode whitespaces from it, so that `postTitle || "Untitled"` actually works even if the title is defined but not visible.

### CODE REVIEW

**Comment by drw158** [comment]
I was able to reproduce by adding blank line above the image in the post.

Hm this branch doesn't seem to fix it :(

**Comment by drw158** [comment]
But if I do add a space in the title this branch does show "Untitled" in the post view

screen shot 2017-11-06 at 3 12 08 pm

File: client/my-sites/comments/comment-list/comment-list-header.jsx [hunk 1]
```diff
@@ -47,7 +47,9 @@ export const CommentListHeader = ( {
 				actionText={ translate( 'View Post' ) }
 				backHref={ `/comments/all/${ siteSlug }` }
 			>
-				<div className="comment-list__header-title">{ postTitle }</div>
+				<div className="comment-list__header-title">
+					{ postTitle.trim() || translate( 'Untitled' ) }
```
Comment (gwwar) [inline client/my-sites/comments/comment-list/comment-list-header.jsx]: Any reason not to update postTitle in mapStateToProps?

**Comment by gwwar** [comment]
👍 Looks pretty close. I verified Dave's case. Looks like if there's an end-line + image, we have a missing title, and we try to render a nbsp.

screen shot 2017-11-06 at 3 28 02 pm


screen shot 2017-11-06 at 3 24 32 pm

File: client/my-sites/comments/comment-list/comment-list-header.jsx (hunk 1 above)
Comment (Copons) [reply client/my-sites/comments/comment-list/comment-list-header.jsx]: None whatsoever, just wanted to be symmetrical with the other file. 🙂

**Comment by Copons** [comment]
I keep not being able to replicate it. :(

@gwwar what do you mean with "end-line"?
I tried by adding a `<p>` before the image (which automatically adds a space in the empty line), and it gets replaced by "Untitled" correctly.
I tried by adding a `<br>` (with `shift+enter`) but it gets stripped out upon post save.

**Comment by gwwar** [comment]
In the Calypso editor press <kbd>enter</kbd> then insert an image.

See https://gwwartest.wordpress.com/2017/11/06/1529/ for an example

**Comment by Copons** [comment]
Yup, that's what I did here: https://coponstestfree.wordpress.com/2017/11/06/1732/

And this is what I see in Calypso on this branch:

screen shot 2017-11-08 at 11 01 56

**Comment by gwwar** [comment]
@Copons I invited you to my test site, so you can more easily repro.

**Comment by gwwar** [comment]
Verified that this is fixed in the new design _only_. 😄

**Pull request merged**